### ISSUE TITLE
Failed to parse the data set imported by the image search system

### ISSUE THREAD

**Issue by want-fly** [milvus-io/bootcamp#641]
**Describe the issue**
根据教程 [图片搜索教程](https://github.com/milvus-io/bootcamp/tree/master/solutions/reverse_image_search)  部署的图片检索系统前端点击无反应 查看`milvusbootcamp/img-search-server:2.0`的日志显示图片已经在解析了快要结束的时候解析报错
**To Reproduce**
1. 机器配置 腾讯云 四核8g
2. 根据教程 [Milvus](https://milvus.io/cn/docs/v2.0.0/install_standalone-docker.md)部署的单机版
3. 采用docker的方式部署的[server ](https://github.com/milvus-io/bootcamp/tree/master/solutions/reverse_image_search/object_detection#21-run-server-with-docker)
4.  [数据集]( https://drive.google.com/file/d/1n_370-5Stk4t0uDV1QqvYkcvyV8rbw0O/view?usp=sharing)
5.  数据集解压到 **/data/image_path/JPEGImages**
6.  前端页面导入数据
![milvus-pic-issue-1](https://user-images.githubusercontent.com/36727480/132441781-155eb9e1-1c2a-46d4-9ad7-8b82c06f64c7.png)
7. 查看 `img-search-server:`的日志出现以下报错
```shell
Extracting feature from image No. 7161 , 7819 images in total
2021-09-08 02:58:14,365 ｜ ERROR ｜ load.py ｜ extract_features ｜ 37 ｜  Error with extracting feature from image cannot identify image file <_io.BytesIO object at 0x7fe2bff07620>
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/app/src/operations/load.py", line 30, in extract_features
    norm_feat = model.resnet50_extract_feat(img_path)
  File "/app/src/encode.py", line 20, in resnet50_extract_feat
    img = image.load_img(img_path, target_size=(self.input_shape[0], self.input_shape[1]))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/preprocessing/image.py", line 296, in load_img
    target_size=target_size, interpolation=interpolation)
  File "/usr/local/lib/python3.6/dist-packages/keras_preprocessing/image/utils.py", line 114, in load_img
    img = pil_image.open(io.BytesIO(f.read()))
  File "/usr/local/lib/python3.6/dist-packages/PIL/Image.py", line 2968, in open
    "cannot identify image file %r" % (filename if filename else fp)
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7fe2bff07620>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/uvicorn/protocols/http/h11_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.6/dist-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/fastapi/applications.py", line 201, in __call__
    await super().__call__(scope, receive, send)  # pragma: no cover
  File "/usr/local/lib/python3.6/dist-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/middleware/cors.py", line 86, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/usr/local/lib/python3.6/dist-packages/starlette/middleware/cors.py", line 142, in simple_response
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.6/dist-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/usr/local/lib/python3.6/dist-packages/fastapi/routing.py", line 215, in app
    dependant=dependant, values=values, is_coroutine=is_coroutine
  File "/usr/local/lib/python3.6/dist-packages/fastapi/routing.py", line 149, in run_endpoint_function
    return await dependant.call(**values)
  File "main.py", line 68, in load_images
    total_num = do_load(item.Table, item.File, MODEL, MILVUS_CLI, MYSQL_CLI)
  File "/app/src/operations/load.py", line 54, in do_load
    vectors, names = extract_features(image_dir, model)
  File "/app/src/operations/load.py", line 38, in extract_features
    sys.exit(1)
SystemExit: 1
2021-09-08 02:58:14,365 ｜ ERROR ｜ h11_impl.py ｜ run_asgi ｜ 399 ｜ Exception in ASGI application
Traceback (most recent call last):
  File "/app/src/operations/load.py", line 30, in extract_features
    norm_feat = model.resnet50_extract_feat(img_path)
  File "/app/src/encode.py", line 20, in resnet50_extract_feat
    img = image.load_img(img_path, target_size=(self.input_shape[0], self.input_shape[1]))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/preprocessing/image.py", line 296, in load_img
    target_size=target_size, interpolation=interpolation)
  File "/usr/local/lib/python3.6/dist-packages/keras_preprocessing/image/utils.py", line 114, in load_img
    img = pil_image.open(io.BytesIO(f.read()))
  File "/usr/local/lib/python3.6/dist-packages/PIL/Image.py", line 2968, in open
    "cannot identify image file %r" % (filename if filename else fp)
PIL.UnidentifiedImageError: cannot identify image file <_io.BytesIO object at 0x7fe2bff07620>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/uvicorn/protocols/http/h11_impl.py", line 396, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.6/dist-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/fastapi/applications.py", line 201, in __call__
    await super().__call__(scope, receive, send)  # pragma: no cover
  File "/usr/local/lib/python3.6/dist-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/middleware/cors.py", line 86, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/usr/local/lib/python3.6/dist-packages/starlette/middleware/cors.py", line 142, in simple_response
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.6/dist-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.6/dist-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/usr/local/lib/python3.6/dist-packages/fastapi/routing.py", line 215, in app
    dependant=dependant, values=values, is_coroutine=is_coroutine
  File "/usr/local/lib/python3.6/dist-packages/fastapi/routing.py", line 149, in run_endpoint_function
    return await dependant.call(**values)
  File "main.py", line 68, in load_images
    total_num = do_load(item.Table, item.File, MODEL, MILVUS_CLI, MYSQL_CLI)
  File "/app/src/operations/load.py", line 54, in do_load
    vectors, names = extract_features(image_dir, model)
  File "/app/src/operations/load.py", line 38, in extract_features
    sys.exit(1)
SystemExit: 1
INFO:     114.219.151.73:10393 - "POST /img/load HTTP/1.1" 500 Internal Server Error

```

**Comment by shiyu22** [comment]
I think it's [an issue about Keras](https://stackoverflow.com/questions/31077366/pil-cannot-identify-image-file-for-io-bytesio-object) it shows the error:
```
  File "/usr/local/lib/python3.6/dist-packages/keras_preprocessing/image/utils.py", line 114, in load_img
    img = pil_image.open(io.BytesIO(f.read()))
  File "/usr/local/lib/python3.6/dist-packages/PIL/Image.py", line 2968, in open
    "cannot identify image file %r" % (filename if filename else fp)
```

If you are starting a server with source code, you can add `try accept` to [the code before extract features](https://github.com/milvus-io/bootcamp/blob/master/solutions/reverse_image_search/quick_deploy/server/src/operations/load.py#L54).

**Comment by want-fly** [comment]
我启动是用docker启动的。而且我发现不管图片有多少 都会在快要解析完的时候结束的时候 报错

**Comment by ss892714028** [comment]
> 我启动是用docker启动的。而且我发现不管图片有多少 都会在快要解析完的时候结束的时候 报错


Are you using your own images or the images provided by the bootcamp?

**Comment by want-fly** [comment]
i use the  images  provided by the bootcamp

**Comment by want-fly** [comment]
> > 我启动是用docker启动的。而且我发现不管图片有多少 都会在快要解析完的时候结束的时候 报错
> 
> Are you using your own images or the images provided by the bootcamp?

我使用的就是文档上提供供的镜像

**Comment by shiyu22** [comment]
I think it may be the image error, so you can also change the code in the docker by run `docker exec -ti <container id>`. Then you can add `try except` to the code before extract features, then you can restart docker by `docker restart <container id>`.

When you change the code, you can also contribute to the repo :)

**Comment by want-fly** [comment]
> I think it may be the image error, so you can also change the code in the docker by run `docker exec -ti <container id>`. Then you can add `try except` to [the code](https://github.com/milvus-io/bootcamp/blob/master/solutions/reverse_image_search/quick_deploy/server/src/operations/load.py#L54) when extract features, then you can restart docker by `docker restart <container id>`.
> 
> When you change the code, you can also contribute to the repo :)

谢谢 我终于成功了  我修改如下代码
![image](https://user-images.githubusercontent.com/36727480/133188420-105f8ffa-8de4-46e0-a157-a0c96a7a1eda.png)

**Comment by shiyu22** [comment]
closed by #646
